### PR TITLE
modules: lvgl: Fix inverting x/y if screen is rotated

### PR DESCRIPTION
**Pull request by glenn-andrews**
https://github.com/zephyrproject-rtos/zephyr/pull/70541 has an issue where if the screen has been rotated, values calculated if invert-x or invert-y are set will be overwritten.

This breaks the adafruit_2_8_tft_touch_v2 touchscreen as the display is rotated by 90 degrees but uses invert-x and invert-y.

This change makes the invert-x and invert-y options independent of screen rotation.

### CODE REVIEW

**Review by faxe1008 — COMMENTED** [review]
Changes make sense, indeed the settings you mentioned have a regression. Three things that I see here:

- With the addition of another temp variable it gets kinda hard to follow the code `data->point_x`, `point_x` and `point->x`. Maybe you have an idea to make it easier to reason about.

- This needs some kind of unit test imo. It's tricky to wrap one's head around and there are also so ambigouities around how all the flags interact with each other (does the swap occur at the start or the end?). A test would document the indended behaviour.

- We need to also take a look at the kscan input driver wrapper. I'd like to deprecate it at some point and intree there is no usage that I am aware of. I suppose this would follow the usual deprecation timeline, right @fabiobaltieri ?

@glenn-andrews Can you maybe try and address the first point? Thanks.

**Comment by glenn-andrews** [comment]
> * With the addition of another temp variable it gets kinda hard to follow the code `data->point_x`, `point_x` and `point->x`. Maybe you have an idea to make it easier to reason about.

But `data->point_x` is not a temporary variable, it's the `data` field of the device driver struct. I was going to modify it directly, but I don't know if it's required to keep it the same. 

Neither is `point->x` temporary as it is a pointer to a filed within the `data` field of the device driver struct.

I probably should make it an `lv_point_t` rather than two separate coordinates. I'll do so now.

**Review by faxe1008 — APPROVED** [review]
This looks better, thanks :^)

File: modules/lvgl/input/lvgl_pointer_input.c [hunk 1]
```diff
@@ -61,37 +61,49 @@ static void lvgl_pointer_process_event(const struct device *dev, struct input_ev
...
+		point->x = cap->x_resolution - tmp_point.x;
+		point->y = cap->y_resolution - tmp_point.y;
+		break;
+	case DISPLAY_ORIENTATION_ROTATED_270:
+		point->x = cap->x_resolution - tmp_point.y;
+		point->y = tmp_point.x;
+		break;
+	default:
+		LOG_ERR("Invalid display orientation");
+		break;
```
Comment (pdgendt) [inline modules/lvgl/input/lvgl_pointer_input.c:106]: Maybe add an assert?

**Review by pdgendt — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by larsgk — APPROVED** [review]
Seems to fix it for me (Adafruit 2.8 TFT, nrf5340 Audio DK)
